normandy/recipes/tests/__init__.py

```diff
@@ -156,6 +160,74 @@ class Meta:
     recipe = factory.SubFactory(RecipeFactory)
     identicon_seed = FuzzyIdenticonSeed()
     comment = FuzzyUnicode()
+    extra_filter_expression = factory.fuzzy.FuzzyChoice(["true", "false"])
+
+    @factory.lazy_attribute
+    def filter_object_json(self):
+        filters = [
+            ChannelFilterFactory(),
+            LocaleFilterFactory(),
+            CountryFilterFactory(),
+            StableSampleFilterFactory(),
+            BucketSampleFilterFactory(),
+        ]
+        return json.dumps(filters)
```

Comment:
So now every recipe revision as a full filter object with all of these? Always.

Reply:
Yes it does:

```python
recipe = RecipeFactory()
pprint(recipe.filter_object)
print(repr(recipe.filter_expression))
```
That prints:
```
[{'channels': ['beta'], 'type': 'channel'},
 {'locales': ['sv'], 'type': 'locale'},
 {'countries': ['SE'], 'type': 'country'},
 {'input': ['normandy.userId', 'normandy.recipe.id'],
  'rate': 0.116283026264581,
  'type': 'stableSample'},
 {'count': 4662,
  'input': ['normandy.userId', '42'],
  'start': 3352,
  'total': 6634,
  'type': 'bucketSample'}]
'(normandy.channel in ["beta"]) && (normandy.locale in ["sv"]) && (normandy.country in ["SE"]) && ([normandy.userId,normandy.recipe.id]|stableSample(0.116283026264581)) && ([normandy.userId,42]|bucketSample(3352,4662,6634)) && (false)'
```

I think it's a mistake to start with everything and explicitly select those that get less (i.e. `RecipeFactory(..., filter_object_json=None)`)

It's quite a fundamental thing and I don't have a link or specific quote to point at but I think fixtures are best kept short, simple, and minimal. That's just experience talking and, I admit, perhaps it's personal preference.

Reply:
I would rather it was like this:

```python
recipe = RecipeFactory()
assert recipe.filter_expression == 'false'
add_filter_object(recipe, ChannelFactory())  # or something similar
assert recipe.filter_expression == '(normandy.channel in ["beta"]) && (false)'
```
(note: the use of `assert` isn't a test assertion, just a demonstration)